modules/ROOT/pages/syntax/keywords.adoc

```diff
@@ -0,0 +1,361 @@
+:description: This section contains a list of reserved keywords in Cypher.
+
+[[cypher-keywords]]
+= Keywords
+
+Keywords are words that have a special meaning in Cypher.
+
+The keywords are not recommended to be used as identifiers in the following contexts:
```

Comment:
is the recommendation strong enough here? if it doesn't work at all, we could say that here and then offer the use of quotes below as a way of still using the keyword as an identifier, if the user wants to do that (if so, remove "unquoted" from line 14)

Reply:
How do you mean strong enough? I am not sure I follow.

After all it is a recommendation. User do not have to do that but it will make their live a little less complicated. The only true safe thing are quote identifiers. But Valerio seems not to be in favour of recommending that. Hence, I only hint at it and leave to the reader make this last step of logical inference.

Reply:
in the contexts pointed out, it only works when using quotations because otherwise they would be interpreted as the keywords, correct?

in a sense, it's not a recommendation then, rather it just doesn't work without quotations

Reply:
Yes. In the specific contexts of ambiguity, which are more subtle then what is listed here, you have to use quotation to get identifier interpretation. 

However, the recommendation is deliberately more coarse grained, recommend to us quotes on identifiers (that are labels, relationship types, and variable) equal to keywords in all places.

For instance, the commendation is to not do something like this:
```
WITH 1 AS RETURN
RETURN RETURN
```
although is works perfectly fine.

As the example hopefully illustrates, the recommendation is more coarse grained than functionally needed, because 
1. it can be confusing for a query reader if you use keywords as identifiers without quotes even if there is no ambiguity for the system
2. you do not have to worry about in where and what these exact contexts of actual ambiguity are, i.e. the recommendation is much simpler and more stable of time

BTW: 
I removed functions names from the recommendation, but maybe we should add that back in, because the following is likewise confusing:
```
WITH WITH("foo") AS RETURN
CALL CALL("bar") YIELD YIELD
RETURN RETURN, YIELD
```
My thinking was that most users do not assign function names. Only if you do have custom functions (UDF) you assigned function names. Nevertheless. I will add function name and procedure them back in.

Reply:
It also included parameters, originally. These can never be ambiguous because they are prefixed with a `$`.

So do we think that 
```
WITH $WITH AS foo
RETURN $RETURN, $AS, foo
```

is to be avoided? I am open to opinions here.

Note that these are deliberately arcane examples.

There are many instance that are less arcane and confusing, e.g.:
```
WITH $USER AS foo
RETURN $TEXT, foo
```
or (without parameters)
```
WITH "bob" AS USER, "Hello" AS START
RETURN START + USER AS SHOW
```

I guess, we do not want to scare user too much about the need of quoting. But that want something to point to if they start to complain about too weird cases — which most users will avoid their own sake.

Reply:
alright, let's leave it as is then :) thanks for taking the time to elaborate